packages/langium/src/grammar/type-system/type-validator.ts

```diff
@@ -217,26 +340,14 @@ function checkPropertiesConsistency(inferred: Property[], declared: Property[],
     }
 
     // detects lack of properties
-    for (const expectedProperty of declared) {
-        const foundProperty = inferred.find(e => expectedProperty.name === e.name);
-        if (!foundProperty && !expectedProperty.optional) {
-            errorToRuleNodes(`A property '${expectedProperty.name}' is expected`);
+    for (const [property, expectedProperties] of declared.entriesGroupedByKey()) {
+        const foundProperty = inferred.get(property);
+        if (foundProperty.length === 0 && !expectedProperties.some(e => e.optional)) {
+            errorToRuleNodes(`A property '${property}' is expected`);
         }
     }
 }
 
-function checkSuperTypesConsistency(inferred: string[], declared: string[], errorToRuleNodes: (error: string) => void): void {
```

Comment:
Removing super types consistency check doesn't allow to notice cases, when parser rules hierarchy doesn't repeat the types hierarchy. The issue example has the types hierarchy: `Foo` <- `Event` and the rules hierarchy: `Event`.
```
interface Foo {
    name: string
}

interface Event extends Foo {}

Event returns Event:
    name=ID;
```
Maybe we don't always want to have a "full match," but it might be a good idea to leave this check and enable it with some kind of flag.